ui/framework/src/ui-framework/stagepanels/StagePanelDef.tsx

```diff
@@ -198,28 +201,25 @@ export class StagePanelDef extends WidgetHost {
 
   /** Gets the list of Widgets. */
   public get widgetDefs(): ReadonlyArray<WidgetDef> {
-    if (this.panelZones) {
-      const widgetDefs = [];
-      for (const [, panelZone] of this.panelZones) {
-        widgetDefs.push(...panelZone.widgetDefs);
-      }
-      return widgetDefs;
+    const widgetDefs = [...super.widgetDefs];
+    for (const [, panelZone] of this.panelZones) {
+      widgetDefs.push(...panelZone.widgetDefs);
     }
-    return super.widgetDefs;
+    return widgetDefs;
   }
 
-  /** Finds a [[WidgetDef]] based on a given id */
-  public findWidgetDef(id: string): WidgetDef | undefined {
-    // istanbul ignore if
-    if (this.panelZones) {
-      for (const [, panelZone] of this.panelZones) {
-        const widgetDef = panelZone.findWidgetDef(id);
-        if (widgetDef)
-          return widgetDef;
-      }
-    }
+  /** Gets the list of Widgets (w/o StagePanelZone widgets).
+   * @internal
+   */
+  public get panelWidgetDefs() {
+    return super.widgetDefs;
+  }
 
-    return super.findWidgetDef(id);
+  /** @internal */
+  public updateDynamicWidgetDefs(stageId: string, stageUsage: string, location: ZoneLocation | StagePanelLocation, _section?: StagePanelSection): void {
+    this.panelZones.start.updateDynamicWidgetDefs(stageId, stageUsage, location, StagePanelSection.Start);
+    // this.panelZones.middle.updateDynamicWidgetDefs(stageId, stageUsage, location, StagePanelSection.Middle);

```

Comment:
Why are middle and end zones commented out?

Reply:
Most of the current consumers ignore section parameter and will return "same" widget definitions for all the section types. We need to have some sort of mechanism to avoid adding multiple widgets with the same widget id first.

Reply:
Shouldn't we add a // TODO comment to this code that states this so we don't forget get to do it and to explain why the lines are commented out?

Reply:
Good idea, comment added.